trusted-dealer/src/cli.rs

```diff
@@ -1,14 +1,60 @@
-use frost_core::keys::IdentifierList;
-use frost_core::Ciphersuite;
 use rand::thread_rng;
+use std::collections::BTreeMap;
 use std::io::{BufRead, Write};
 
+use frost_core::keys::{IdentifierList, PublicKeyPackage, SecretShare};
+use frost_core::{Ciphersuite, Identifier};
+use reddsa::frost::redpallas::keys::EvenY;
+
 use crate::args::Args;
 use crate::inputs::{print_values, request_inputs};
 use crate::trusted_dealer_keygen::{split_secret, trusted_dealer_keygen};
 
-// Currently this uses the Default Identifiers
-pub fn cli<C: Ciphersuite + 'static>(
+// The redpallas ciphersuite requires ensuring public key have an even Y
+// coordinate. Since the code uses generics, this trait is used to convert if
+// needed depending on the ciphersuite.
```

Comment:
RedPallas does not require this; the signature scheme's public keys are the set of Pallas group elements, which does not exclude odd Y coordinates ([spec](https://zips.z.cash/protocol/protocol.pdf#concretereddsa)). The requirement for even Y is specifically on the Orchard Spend validating key `ak` ([spec](https://zips.z.cash/protocol/protocol.pdf#orchardkeycomponents)) (in order for `Extract_P` to not lose information).

Reply:
clarified in e283b9a4